dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/citation/CitationDataExtractorTest.java

```diff
@@ -35,6 +38,26 @@ void create() throws ParseException {
         assertThat(citationData.getYear()).isEqualTo("1955");
     }
 
+    @Test
+    @DisplayName("Should extract data for citation from harvested dataset")
+    void create_forHarvested() throws ParseException {
+
+        // given & when
+        DatasetVersion datasetVersion = utils.createATestDatasetVersion("Dataset Title", true);
+        datasetVersion.getDataset().setHarvestedFrom(new HarvestingClient());
+        CitationData citationData = dataExtractor.create(datasetVersion);
+
+        // then
+        assertThat(citationData.getAuthorsString()).isEqualTo("First Last");
+        assertThat(citationData.getFileTitle()).isNull();
+        assertThat(citationData.getPersistentId().asString()).isEqualTo("doi:10.5072/FK2/LK0D1H");
+        assertThat(citationData.getPublisher()).isNull();
+        assertThat(citationData.getTitle()).isEqualTo("Dataset Title");
+        assertThat(citationData.getUNF()).isNull();
+        assertThat(citationData.getVersion()).isNull();
+        assertThat(citationData.getYear()).isEqualTo(String.valueOf(Year.now().getValue()));
```

Comment:
Maybe we could add a test that will have `.getYear` taken from `ProductionDate` metadataField ?

Reply:
I'm guessing you mean `DistributionDate`.